FirebaseAdmin/FirebaseAdmin/Messaging/FirebaseMessagingClient.cs

```diff
@@ -157,7 +152,7 @@ internal static string ClientVersion
             }
             catch (HttpRequestException e)
             {
-                throw new FirebaseException("Error while calling the FCM service.", e);
+                throw e.ToFirebaseException();
```

Comment:
It's a little weird to me that the messaging client won't always throw an instance of `FirebaseMessagingException`. I mean, I understand that there's no extra information to be included. But it seems like people might expect it to always throw the same type. And `FirebaseMessagingException` does support the messaging code being null. Thoughts?

Reply:
Good point. Actually I meant to throw only `FirebaseMessagingException` from this API, but missed that requirement in the implementation. I've fixed it now. Also updated the API reference in `FirebaseMessaging` and added a test case.